flink-scala/src/main/scala/org/apache/flink/api/scala/DataSet.scala

```diff
@@ -1599,7 +1601,77 @@ class DataSet[T: ClassTag](set: JavaDataSet[T]) {
   def output(outputFormat: OutputFormat[T]): DataSink[T] = {
     javaSet.output(outputFormat)
   }
-  
+
+  /**
+    * Selects an element with minimum value.
+    * <p>
+    * The minimum is computed over the specified fields in lexicographical order.
+    * <p>
+    * <strong>Example 1</strong>: Given a data set with elements <code>[0, 1], [1, 0]</code>, the
+    * results will be:
+    * <ul>
+    * <li><code>minBy(0)</code>: <code>[0, 1]</code></li>
+    * <li><code>minBy(1)</code>: <code>[1, 0]</code></li>
+    * </ul>
+    * <p>
+    * <strong>Example 2</strong>: Given a data set with elements <code>[0, 0], [0, 1]</code>, the
+    * results will be:
+    * <ul>
+    * <li><code>minBy(0, 1)</code>: <code>[0, 0]</code></li>
+    * </ul>
+    * <p>
+    * If multiple values with minimum value at the specified fields exist, a random one will be
+    * picked.
+    * <p>
+    * Internally, this operation is implemented as a {@link ReduceFunction}.
+    *
+    */
+  def minBy(fields: Array[Int]) : ReduceOperator[T]  = {
```

Comment:
`fields: Int*` instead of `fields: Array[Int]` (see groupBy(fields: Int*)`. 
Do not return a Java API ReduceOperator. This will mix Java and Scala DataSet API which must not happen! Return `DataSet[T]` instead.